### DISCUSSION TITLE
How I Got Started with GitHub :rocket:

### DISCUSSION THREAD

**Discussion by queenofcorgis** [community/community#32792] [Discover: GitHub Best Practices]
Not long ago (a little over eight months ago to be exact), I was a new GitHub user myself. Before I joined GitHub, the only thing I had done was set up my profile. My contribution graph was very blank. Not for long though 🟩 as you may know we at GitHub do a large part of our work on GitHub every day, so I dove into the challenge of learning this new tool. 

The main thing I learned along the way was there isn’t a one-size-fits-all approach to learning GitHub, so some resources may be better suited for you than they were for me and vice versa. 

#### 🧪 Learning Lab ➡️ ✏️ GitHub Skills 

The best place to start teaching yourself GitHub is unsurprisingly on GitHub! When I started we had Learning Lab which has now evolved into the Actions-backed [GitHub Learn](https://skills.github.com/). It was a happy medium between a DIY approach and having an actual teacher. I was able to go at my own pace, but when I ran into snags (more than I’d care to admit) the bot suggested ways I may have erred and how to get back on track. This helped build my confidence and get the lay of the land. 

Now with GitHub Skills you have programs for your first day, first week, and beyond to get comfortable with using the platform.

#### 🏫 Courses Around the Web
There are no shortage of classes, tutorials, and guides online for you to choose from in your learning journey. While there are many similarities with programs like Codeacademy, Udemy, and more, I found that it was good reinforcement to learn these foundational skills from different perspectives. 

#### 💬 Community on Discussions

The Community isn’t just a place to go to when you’ve got a specific question, it is a space filled with users just like you, especially in the [New to GitHub](https://github.com/orgs/community/discussions/categories/new-to-github) category. No question is too small or too big here and you’ll find other members who are starting out as well. 

My *number one tip for learning GitHub* was trying to solve other’s questions, even if they had already been solved. Doing this took me to far corners of the GitHub platform that may otherwise have taken me ages to find, learn to troubleshoot myself, and read through our [docs](https://docs.github.com/) 👓. 

But just poking around interesting Community threads was massively helpful as well, to learn terminology, how folks use GitHub, and view their projects. So take a look 👀 around, your next learning opportunity might be just a few clicks away. 

####  📃 Cheatsheets 
Having handy guides bookmarked so I could easily find foundational information for coding, navigating GitHub, and using Markdown made me feel much more comfortable. There is so much to absorb in the early days of GitHub and it can be extremely overwhelming when you feel like you have to memorize everything being thrown at you. 

Some cheatsheets for you: 
* [Git Cheat Sheet](https://education.github.com/git-cheat-sheet-education.pdf) 
* [Markdown Guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet) 

Using a variety of resources, learning platforms, and the Community was crucial to my development, but I’m still learning 🎒 as are many of us here. Have any resources or methods you use when learning GitHub, let us know below. 

Looking to make sure you’ve really developed beginner skills? Don’t forget about our [New User Checklist](https://github.com/orgs/community/discussions/27367). Happy learning 🌈!

**Comment by Thanhhoang19932** [comment]
Yhb

**Comment by jefflowkey** [comment]
Which course under github skills did you start with and which one do you recommend???

**Reply by queenofcorgis** [reply]
Hey there, welcome to GitHub, glad you're ready to dive in! I started with _Introduction to GitHub_ as I wanted a core foundation to build off of. I recommending spending some time with _Review Pull Requests_ and _Connect the Dots_ as those are both key skills to have for using GitHub. But depending on your skillset and what you'd like to achieve there is a whole section dedicated GitHub Actions as well. I can't wait to hear how it goes and what you think!

**Reply by jefflowkey** [reply]
Thanks for the insight.
I have been using github for a whole but for basically the simple stuff.
Create a repo and use it to save my code.
Add a few branches and do some pr.

I know there's more and would really want to be know more

**Comment by kashirk** [comment]
One thing that my seniors advised me to avoid feeling overwhelmed was creating a separate "sandbox" repository where I could freely experiment.

I practiced branching, rebasing, resolving merge conflicts, writing Markdown, setting up GitHub Actions, and even intentionally making mistakes just to understand how to recover from them. Having a place where nothing could break made learning much less intimidating.

Learning how to use Issues for planning, Projects for tracking work, Discussions for knowledge sharing, Pull Requests for code reviews, and Actions for automation gives a much better understanding of the platform as a complete collaboration tool. Even in solo projects, using these features builds habits that transfer well to team environments later on.

Thanks for sharing these resources—they make the learning path much less overwhelming for new users.
